databricks_cli/secrets/cli.py

```diff
@@ -0,0 +1,312 @@
+    contains all users in the workspace. If "initial_manage_principal" is not specified,
+    the initial ACL with MANAGE permission applied to the scope is assigned to the
+    API request issuer's user identity.
+    """
+    if initial_manage_principal is not None and initial_manage_principal != "users":
+        error_and_quit("The only supported principal for --initial-manage-principal is users")
+    SecretApi(api_client).create_scope(scope, initial_manage_principal)
+
+
+def _scopes_to_table(scopes_json):
+    ret = []
+    for s in scopes_json.get('scopes', []):
+        ret.append((truncate_string(s['name']), s['backend_type']))
+    return ret
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Lists all secret scopes.')
+@click.option('--output', help=OutputClickType.help, type=OutputClickType())
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_scopes(api_client, output):
+    """
+    Lists all secret scopes.
+    """
+    scopes_json = SecretApi(api_client).list_scopes()
+    if OutputClickType.is_json(output):
+        click.echo(pretty_format(scopes_json))
+    else:
+        click.echo(tabulate(_scopes_to_table(scopes_json), headers=SCOPE_HEADER))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Deletes a secret scope.')
+@click.option('--scope', required=True, type=SecretScopeClickType(), help=SecretScopeClickType.help)
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_scope(api_client, scope):
+    """
+    Deletes a secret scope.
+    """
+    SecretApi(api_client).delete_scope(scope)
+
+
+def _verify_and_translate_options(string_value, binary_file):
+    """
+    Translates options into actual parameters for API call.
+    Return tuple with two values representing (string_value, bytes_value).
+    """
+    if string_value and binary_file:
+        error_and_quit("At most one of {} should be provided."
+                       .format(['string-value', 'binary-file']))
+
+    elif string_value is None and binary_file is None:
+        prompt = '# Remove this line and input your secret value.' + \
+            ' Any trailing new line will be stripped and text will be stored in' + \
+            ' UTF-8 (MB4) form. Exit without saving to abort writing secret.'
+
+        # underlying edit function made sure using a temporary file for editing
+        content = click.edit(prompt)
+        # return None means editor is closed without changes
+        if content is None:
+            error_and_quit('No changes made, write secret aborted.'
+                           ' Please follow the instruction to input secret value.')
+
+        return content.rstrip('\n'), None
+
+    elif string_value is not None:
+        return string_value, None
+
+    elif binary_file is not None:
+        with open(binary_file, 'rb') as f:
+            binary_content = f.read().rstrip('\n')
+
+        if not is_base64_str(binary_content):
```

Comment:
We shouldn't expect base64 encoded files. Instead, we should base64-encode the content and send it to the endpoint.